.github/workflows/e2e-upgrade.yaml

```diff
@@ -1,5 +1,15 @@
 name: E2EUpgrade
 on:
+  workflow_dispatch:
+    inputs:
+      git_repo:
+        type: string
+        default: "aws/karpenter"
+      from_git_ref:
+        type: string
+        required: true
+      to_git_ref:
+        type: string
```

Comment:
Shouldn't this be required or at least defaulted to HEAD?

Reply:
If not specified, it does default to an empty string and will be on HEAD